app/src/main/java/net/squanchy/notification/NotificationCreator.kt

```diff
@@ -0,0 +1,203 @@
+package net.squanchy.notification
+
+import android.annotation.TargetApi
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.BitmapFactory
+import android.graphics.Color
+import android.os.Build
+import android.support.v4.app.NotificationCompat
+import android.support.v4.app.TaskStackBuilder
+import android.support.v4.content.ContextCompat
+import android.text.TextUtils
+import net.squanchy.R
+import net.squanchy.eventdetails.EventDetailsActivity
+import net.squanchy.home.HomeActivity
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.speaker.domain.view.Speaker
+import net.squanchy.support.android
+import net.squanchy.support.lang.Lists.map
+import java.util.ArrayList
+
+class NotificationCreator(private val context: Context) {
+
+    fun createFrom(events: List<Event>): List<Notification> {
+        if (android.isAtLeastOreo) {
+            createChannel()
+        }
+
+        val notifications = events.mapTo(ArrayList()) { createFrom(it) }
```

Comment:
I think we can use the normal `map` here, the resulting list is a different one so no side effects anyway

Reply:
Unfortunately not, unless I'm missing something, because using `map` will only return a `List`, while we need a mutable one to be able to add the summary one, of needed, a few lines later

Reply:
Oh I see, nice!